### DISCUSSION TITLE
Shotcut Proposal for switching side menü and timeline player

### DISCUSSION THREAD

**Discussion by flowerhorse** [brunoherbelin/vimix#88] [Ideas]
If I am on the F1 screen and select a medium there, this can have two effects on the page menu: Either the page menu is closed, in which case only the highlight bar at the file abbreviation moves, or the menu is open, in which case the color correction information is also displayed. Double-clicking generally opens the side menu and generally turns it off the second time.

The work with the timeline player could also be an important reason but this point has to be selected separately so far, which again costs time.
My suggestion would be, already with 'onselect' (one cklick or selection by frame? (Is this possible, because I can also couple two objects with the selection window?) generally open the side menu and open the timeline player with a double click. The second double click on the same object could then collapse the side menu and the player. This would reduce the switching time and thus improve the live responsiveness.

**Comment by brunoherbelin** [comment]
I see your point and the imagined mechanism was not made explicit and not shown to the user; thanks again for pointing out these limitations in UX. Please let me know if the following seems ok.

Clic mechanism:
- single clic on a source makes it the current source (selected)
    - changing current source if the side panel is open: change source properties
- double clic on a source opens the side panel ('menu') to show its properties
- clic out of a source (clic in background) unsets the current source and hides the side panel.

Clic in current source:
- clic on symbol of the source (top right corner) : opens the source Player
- clic on the initials of the source (top left corner) : opens the side panel (equivalent to double clic) [NEW]
- CTRL + clic on the lock icon : lock/unlock the source

and [NEW] ; mouse over current source :  icons change color to show they are active areas:

![Screenshot from 2023-04-16 23-22-01](https://user-images.githubusercontent.com/60393189/232343100-d55786c4-1d30-4c54-bf4c-b54fcb73d70c.png)

**Comment by flowerhorse** [comment]
My suggestion would be to make the whole thing 'gross motor' and intuitively reachable. So only with very general target points, because that makes the process faster:
click on source: selects AND opens medium properties in left menu
double click on source - opens medium in player
doubleclick if player open or cross clicking - closes player
ADDED:  doubleclick in any passive part of left menu - closes left menu if open

Doubleclick File Abbreviation
I never used the file abbreviation as click-destination because it is too cumbersome for me with the expanding and collapsing of the menu. When I double-click source X in general now, the edit selection currently opens immediately. When I click the file shortcut, I first open or close the side menu. If I double-click to close it after editing and then select a new source, I have to click the shortcut a second time to reopen the editing view. This is an unnecessary interruption of the process. 
Solution: one click on source: select AND opens left menu. There I have immediately everything I need. If the side menu bothers me I click 'Doubleclick in any part of left menu' and it folds. Everything without thinking, in passing. That's how I think it should be for basic functions.

> Mouse over current source : icons change color to show they are active areas:

I don't think it's bad, but there are too many media types where I can't use the player at all. 
That makes the switch more important than the effect of switching:

Doubleclick the Media symbol
Now that I know i can reach the player by clicking mediatype symbol , it makes sense to me. At the beginning I didn't think of it because I only thought of the symbol as a general media type information. What I find difficult is that the media type can be different: If I open a 'picture' with a double click on the picture symbol in the player, it doesn't make sense, because I can't set anything there. Likewise with 'generated media'. Here it says 'Nothing to play' in the player. What I want to say is that the cases that result from 'click media symbol' are not logically comprehensible because sometimes they don't make sense. What should I do with a picture in the player if I can't set anything and don't even have a timeline? Also for this reason I would prefer 'Doubleclick on source - opens medium in source player'.

I think that the functions you work with most often should be accessible as soon as possible. I will also examine the program later to see which functions are the hardest to reach.

**Comment by brunoherbelin** [comment]
Okay, let's establish a clear and smooth process for source selection and panels activation!  As you say, it is important to avoid interruptions and at the same time keep the spontaneity to bring the info when needed !  And yes, I agree with you that, as it is now, it is not optimal. I have changed this several times in the development of vimix and the best solution is not yet achieved.

There are however constraints:
- the Player cannot be pop-upped only on clic on a source: it;s also used for showing multiple files, and has the 'Batch' mode that shows a prepared list of sources; in this case, it could be displayed sources even if no source is selected. Showing the player can be used to see the timing of some sources while acting on different selected one. For this reason, the Player is not a 'property' window of the active source.
- a panel cannot show-up anytime just when clicking on a source: other users ask that the scene remains clear of windows and no side panel while they manipulate sources in the workspace.

So, inline with what you suggest, we could work on the following:

> - click on source: selects AND opens medium properties in left menu

yes, BUT should depend on user's choice : in a previous version of vimix, I had this 'pin' on the left properties: with 'pinned' panel, a single clic was showing the source properties, if 'unpinned', a single clic was only selecting the source (not showing the panel) :

![Screenshot from 2023-04-18 19-01-14](https://user-images.githubusercontent.com/60393189/232852719-bb646f2c-57fb-4712-bb7c-bde331661fe8.png)

This was abandoned for simplicity, but the idea is still there. Would that be useful?

> - double click on source - opens medium in player

OK, it can be done ; could be identical to clicking on the media symbol. 
In fact, I could even sugest that single clic on a source would NOT switch which source is in the player, only a double clic? (i.e. I can watch the player of source A while I manipulate source B) 

> - doubleclick if player open or cross clicking - closes player

not sure...
What you mean is "if the source is current, if it is the one in the player (not a batch), then double clic would close the Player" : that means two conditions that are not obvious. What if I double clic a source but the player is on a Batch? It should not close the Player, but hen how do I associate double-clic to open/close player? 


> - ADDED: doubleclick in any passive part of left menu - closes left menu if open

clic in background of the view? backgound of the panel would be where?


> If I open a 'picture' with a double click on the picture symbol in the player, it doesn't make sense, because I can't set anything there.

Already, the 'Player' is used to show the difference between pre- and post-color correction for any source : thus showing the 'Player' is quite generic in fact. 
+ we'll see with other possibilities to add a timeline to static sources (as discussed in #78),

So, maybe what is true in what you observe is that the term 'Player' is not fully correct. Maybe 'Inspector' ?

**Comment by brunoherbelin** [comment]
Hey there ! 

I'm back on this discussion with a new feature : 

![image](https://github.com/brunoherbelin/vimix/assets/60393189/15414c95-7b24-41c3-837c-cdda11a23c72)

It is now possible to enforce the left panel to be always visible; in this mode, one click on source selects AND opens medium properties in left panel. 

This way, vimix has a mode where the left panel is rapidly available (i.e. maybe more when configuring the session), and another mode with more space (e.g. maybe during performance). 

You can try this in Beta c846a0626f3f507442a3b1c208160a692adb870b

**Comment by brunoherbelin** [comment]
The feature to lock the left panel is now stable in version 0.8.1 (flatpak and snap).
